docs/articles/documentation/extending-testcafe/creating-custom-reporter/index.md

```diff
@@ -0,0 +1,143 @@
+
+Another way to figure out what data and how it can be used in reports is to explore an existing reporter,
+say, the *list* reporter at [https://github.com/DevExpress/testcafe-reporter-list](https://github.com/DevExpress/testcafe-reporter-list).
+You can either run the `preview` Gulp task to view the reporter output in your terminal, or find the etalon report at `/test/data/report-without-colors`.
+
+In the `src/index.js` file, you may also need to implement the [createErrorDecorator](/testcafe/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/#createerrordecorator) method to create a custom error decorator.
+
+```js
+createErrorDecorator () {
+    return {
+        'span category':       () => '',
+        'span step-name':      str => `"${str}"`,
+        'span user-agent':     str => this.chalk.gray(str),
+        'div screenshot-info': str => str,
+        'a screenshot-path':   str => this.chalk.underline(str),
+        'code':                str => this.chalk.yellow(str),
+        'code step-source':    str => this.chalk.magenta(this.indentString(str, 4)),
+        'span code-line':      str => `${str}\n`,
+        'span last-code-line': str => str,
+        'code api':            str => this.chalk.yellow(str),
+        'strong':              str => this.chalk.cyan(str),
+        'a':                   str => this.chalk.yellow(`"${str}"`)
+    };
+},
+```
+
+For more information on implementing the mentioned methods, see [this topic](/testcafe/documentation/extending-testcafe/creating-custom-reporter/reporterpluginhost/#methods-to-implement).
+
+## Workflow Tasks
+
+### Testing Reporter
+
+To test your reporter works well, you can use the `test` Gulp task.
+To get this task to work, you need to provide baseline reports in your format. During testing, reports generated by the custom reporter will be compared with these baseline reports.
+
+You have to compose two baseline reports.
+
+* A regular report. Put it to `/test/data/` and name it `report-without-colors`.
+* A colored report, which is a JSON file that consists of a single string - the report text with additional color information specified using [ANSI escape codes](https://en.wikipedia.org/wiki/ANSI_escape_code). Put this JSON file to `/test/data/` and name it `report-with-colors.json`.
+
+To know how these baseline reports should look, explore the etalon reports of the build-in reporters. You can find the etalon reports in the `/test/data/` directory of each reporter.
+
+After the baseline reports are ready, you can use the `test` task.
+
+```bash
+gulp test
+```
+
+### Preview Report
+
+You can preview a report built by your reporter using the `preview` Gulp task.
+
+```bash
+gulp preview
+```
+
+### Publish Reporter
+
+To publish the reporter at npm, use the `publish` Gulp task.
+
+```bash
+gulp publish
+```
+
+## Installing Reporter
```

Comment:
We need to describe somewhere here how it can be tested with the testcafe, using `npm link` without the need to publish it to the npm.